src/Http/RequestTransformer.php

```diff
@@ -48,6 +48,12 @@ public static function toCake(PsrRequest $request)
             $post = Hash::merge($post, $files);
         }
 
+        $input = null;
+        $stream = $request->getBody();
+        if ($stream->getSize()) {
```

Comment:
This is what I meant:
```php
$input = $stream->geContents();
$input = $input === '' ? null : $input;
```
 My goal is to save the call to `getSize()`, since in many stream implementations that is done by literally counting the bytes one by one in php